### PR TITLE
fixed mae metric bug with lgbm (clean PR)

### PR DESCRIPTION
**Pull request by Zacchaeus14**
Hi @pplonski ,

I checked LightGBM docs and used the following code to check:
```
from lightgbm import LGBMRegressor
import numpy as np
X = np.random.randn(100, 10)
y = np.random.randn(100)
model = LGBMRegressor()
model.fit(X, y, eval_metric="mae")
```
The code excuted sucessfully, which means that LightGBM does support "mae" as metric. However, the problem exist in optuna mode for some reason. LightGBMObjective is returning "l1" instead of "mae"...
Anyways, by mapping "mae" to "l1" the error disappears.

Zacchaues

### CODE REVIEW

**Comment by pplonski** [comment]
Thank you @Zacchaeus14!

I'm thinking that the problem might be when the validation dataset is used. Then the `l1` name is used as a key to the dict with results. The dict with results from LightGBM is used for early prunning in Optuna mode, and for training early stopping. 

As you wrote, mapping to `l1` should fix all issues.
